apps/app/repository.tf

```
@@ -12,4 +12,162 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-// TODO: GH stuff will go here
+resource "github_repository" "app" {
+  name        = var.app_id
+  description = var.app_config.description
+  visibility  = var.app_config.managed_repo.private ? "private" : "public"
+
+  template {
+    owner      = "UCLH-Foundry"
+    repository = var.app_config.managed_repo.template
+  }
+}
+
+resource "github_team" "owners" {
+  name        = "${var.app_id} - owners"
+  description = "Owners of the ${var.app_id} FlowEHR app with push and PR/deployment approval permissions."
+  privacy     = "closed"
+}
+
+resource "github_team_members" "owners" {
+  for_each = var.app_config.owners
+  team_id  = github_team.owners.id
+
+  members {
+    username = each.value
+    role     = "maintainer"
+  }
+}
+
+resource "github_team_repository" "owners_repo_permissions" {
+  team_id    = github_team.owners.id
+  repository = github_repository.app.name
+  permission = "push"
+}
+
+resource "github_repository_file" "codeowners" {
+  repository          = github_repository.app.name
+  branch              = var.environment
+  file                = "CODEOWNERS"
+  content             = <<EOF
+# Owners for branch protection
+# Users within this team are required reviewers for this deployment branch
+*       @${var.github_owner}/${github_team.owners.slug}
+EOF
+  commit_message      = "Add codeowners (managed by Terraform)"
+  commit_author       = "Terraform"
+  commit_email        = "terraform@flowehr.io"
+  overwrite_on_create = true
+}
+
+resource "github_team" "contributors" {
+  name        = "${var.app_id} - contributors"
+  description = "Contributors to the ${var.app_id} FlowEHR app with push permissions."
+  privacy     = "closed"
+}
+
+resource "github_team_members" "contributors" {
+  for_each = var.app_config.contributors
+  team_id  = github_team.contributors.id
+
+  members {
+    username = each.value
+    role     = "member"
+  }
+}
+
+resource "github_team_repository" "contributors_repo_permissions" {
+  team_id    = github_team.contributors.id
+  repository = github_repository.app.name
+  permission = "push"
+}
+
+resource "github_branch" "deployment" {
+  repository = github_repository.app.name
+  branch     = var.environment
+}
+
+resource "github_branch_protection" "deployment" {
+  repository_id       = github_repository.app.name
+  pattern             = var.environment
+  allows_deletions    = false
+  allows_force_pushes = false
+
+  required_status_checks {
+    strict   = true
+    contexts = ["lint"]
```

```suggestion
    contexts = ["Lint"]
```
https://github.com/UCLH-Foundry/Dash-Seedling/blob/1191b390e0ff82aa42ca21d31e173f3d90d026cc/.github/workflows/lint.yml#L1